api/README.md

```diff
@@ -1 +1,269 @@
+
+```javascript
+{
+  "code" : 1234,
+  "message" : "Comment could not be added.",
+  "description" : "The post you are trying to add the comment to was archived."
+}
+```
+
+> In non-production environments the error resource might contain additional fields (e.g. stacktrace, timestamp, etc).
+
+### Casing
+
+To provide better readability all the resource fields should be written in `snake_case`.
+For example:
+```javascript
+{
+  "created_at": 1470191016653,
+  "updated_at": 1470391016653
+}
+```
+
+#### Best practices to name fields
+
+- For dates and timestamps use the `_at` suffix whenever possible, e.g `"created_at": 1470191016653`.
+- For booleans never include `is_` or `has_` prefixes, instead try to use adjectives and verbs in past tense, e.g. `"completed: true"`, `"locked": false`, `"visible": true`, etc.
+- For lists try to use the plural form instead of suffixing it with `_list`, e.g. ~~`category_list`~~ should become `categories`.
+- For resource identifiers use the field `id`. When there is a reference to another resource then prefix it with the name of the relation, e.g. a post would have an author as `author_id`:
+
+```javascript
+{
+  "id": 1,
+  "author_id": 12
+}
+```
+
+## Security and Optimization
+
+### SSL
+
+Use SSL whenever possible. It provides encryption even through unencrypted networks and simplies authentication efforts. Simple API tokens will work just well.
+
+**Be careful!** Once you enabled SSL on your API don't allow unsecured connections (e.g. plain http connections).
+
+> **Note:** Setup SSL on the load-balancer instead of the servers whenever possible.
+
+### Compression
+
+Up to 80% bandwidth saving can be achieved when using gzip compression (according to Twitter and Stack Exchange). Although this requires additional CPU time to (un)compress the results, the trade-off with network costs usually makes it very worthwhile.
+
+Most frameworks/servers provide an easy way to setup gzip compression and most of the time you can also configure a minimum response size which the compression is enabled on. This way you don't overload the CPU with compressing tiny chuncks of responses.
+
+In order to receive a gzip-encoded response you must do two things:
+
+1. Set the `Accept-Encoding` header to gzip
+1. Modify your user agent to contain the string gzip.
+
+Here is an example of properly formed HTTP headers for enabling gzip compression:
+```HTTP
+Accept-Encoding: gzip
+User-Agent: my program (gzip)
+```
+
+> **Note:** Setup compression on the load-balancer instead of the servers whenever possible.
+
+## Version Your API
+
+Make the API version mandatory and do not release an unversioned API. Use a simple ordinal number and avoid dot notation such as 2.5, e.g. `/api/v1/posts`.
```

Comment:
We cannot use semantic versioning in the API since that would change with every modification. The API url contains only the major version, so it will be updated only in case of breaking changes. We will cover this in the document.